pos_mail_receipt/static/src/js/screens.js

```diff
@@ -48,9 +48,16 @@ odoo.define("pos_mail_receipt.screens", function (require) {
             var self = this;
             options = options || {};
             var timeout = typeof options.timeout === 'number' ? options.timeout : 7500;
-            this.$('.button.email').addClass("highlight");
+            var timeout_db_sync = typeof options.timeout_db_sync === 'number' ? options.timeout : 5000;
+            var i = 0
+            var waiting_interval = 1000
             while (self.pos.get("synch").state == "connecting") {
-                await sleep(1000);
+                await sleep(waiting_interval);
+                i = i + waiting_interval
+                if (i >= timeout_db_sync){
+                    self.gui.show_popup('error', _t('Timeout syncronising the database'));
```

Comment:
This popup is not very helpful to the user.

```suggestion
                    self.gui.show_popup('error', _t('Aborted the sending of the e-mail receipt; could not synchronise with the database in time.'));
```

Reply:
Also, add a `self.lock_screen(false)` line here if the `lock_screen(true)` line is moved

Reply:
> This popup is not very helpful to the user.
Is "'Syncronisation with the database is taking too long. Retry or contact the system administrator.' better ? It's hard to write a helpful message since I don't know what actually *is* database syncronisation !


> Also, add a `self.lock_screen(false)` line here if the `lock_screen(true)` line is moved
To be sure I understand, do you mean adding *another* message if the state is `connecting`, before waiting for the timeout ? I am affraid that this could trigger too often...

Reply:
I like your proposal! Just fix the spelling to 'synchronisation' :)

(Database synchronisation, as I understand it, is the browser syncing its local cache with stuff in the DB. But I'm not 100% on the details either.)

Reply:
OK, fixed :)